integrationTests/testConsensusNode.go

```diff
@@ -0,0 +1,462 @@
+		node.WithDataComponents(dataComponents),
+		node.WithStateComponents(stateComponents),
+		node.WithNetworkComponents(networkComponents),
+		node.WithRoundDuration(roundTime),
+		node.WithConsensusGroupSize(consensusSize),
+		node.WithConsensusType(consensusType),
+		node.WithGenesisTime(time.Unix(startTime, 0)),
+		node.WithValidatorSignatureSize(signatureSize),
+		node.WithPublicKeySize(publicKeySize),
+	)
+
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+}
+
+func (tcn *TestConsensusNode) initShardCoordinator(
+	consensusSize int,
+	hasher hashing.Hasher,
+	epochStartRegistrationHandler notifier.EpochStartNotifier,
+	eligibleMap map[uint32][]nodesCoordinator.Validator,
+	waitingMap map[uint32][]nodesCoordinator.Validator,
+	pkBytes []byte,
+	cache storage.Cacher,
+) {
+	argumentsNodesCoordinator := nodesCoordinator.ArgNodesCoordinator{
+		ShardConsensusGroupSize: consensusSize,
+		MetaConsensusGroupSize:  1,
+		Marshalizer:             TestMarshalizer,
+		Hasher:                  hasher,
+		Shuffler:                &shardingMocks.NodeShufflerMock{},
+		EpochStartNotifier:      epochStartRegistrationHandler,
+		BootStorer:              CreateMemUnit(),
+		NbShards:                1,
+		EligibleNodes:           eligibleMap,
+		WaitingNodes:            waitingMap,
+		SelfPublicKey:           pkBytes,
+		ConsensusGroupCache:     cache,
+		ShuffledOutHandler:      &elrondShardingMocks.ShuffledOutHandlerStub{},
+		ChanStopNode:            endProcess.GetDummyEndProcessChannel(),
+		NodeTypeProvider:        &nodeTypeProviderMock.NodeTypeProviderStub{},
+		IsFullArchive:           false,
+		EnableEpochsHandler: &testscommon.EnableEpochsHandlerStub{
+			IsWaitingListFixFlagEnabledField: true,
+		},
+		ValidatorInfoCacher: &vic.ValidatorInfoCacherStub{},
+	}
+
+	tcn.NodesCoordinator, _ = nodesCoordinator.NewIndexHashedNodesCoordinator(argumentsNodesCoordinator)
+}
+
+func (tcn *TestConsensusNode) initBlockChain(hasher hashing.Hasher) {
+	if tcn.ShardCoordinator.SelfId() == core.MetachainShardId {
+		tcn.ChainHandler = CreateMetaChain()
+	} else {
+		tcn.ChainHandler = CreateShardChain()
+	}
+	//_ = tcn.ChainHandler.SetGenesisHeader(&dataBlock.Header{})
+
+	rootHash := []byte("roothash")
+	header := &dataBlock.Header{
+		Nonce:         0,
+		ShardID:       tcn.ShardCoordinator.SelfId(),
+		BlockBodyType: dataBlock.StateBlock,
+		Signature:     rootHash,
+		RootHash:      rootHash,
+		PrevRandSeed:  rootHash,
+		RandSeed:      rootHash,
+	}
+
+	_ = tcn.ChainHandler.SetGenesisHeader(header)
+	hdrMarshalized, _ := TestMarshalizer.Marshal(header)
+	tcn.ChainHandler.SetGenesisHeaderHash(hasher.Compute(string(hdrMarshalized)))
+}
+
+func (tcn *TestConsensusNode) initBlockProcessor() {
+	tcn.BlockProcessor = &mock.BlockProcessorMock{
+		ProcessBlockCalled: func(header data.HeaderHandler, body data.BodyHandler, haveTime func() time.Duration) error {
+			_ = tcn.ChainHandler.SetCurrentBlockHeaderAndRootHash(header, header.GetRootHash())
```

Comment:
in the real component this is actually done inside the CommitBlock part. We should remove it from here